utils/SwiftFloatingPointTypes.py

```diff
@@ -0,0 +1,48 @@
+# ===--- SwiftFloatingPointTypes.py ----------------------------*- coding: utf-8 -*-===//
+#
+# This source file is part of the Swift.org open source project
+#
+# Copyright (c) 2014 - 2016 Apple Inc. and the Swift project authors
+# Licensed under Apache License v2.0 with Runtime Library Exception
+#
+# See http://swift.org/LICENSE.txt for license information
+# See http://swift.org/CONTRIBUTORS.txt for the list of Swift project authors
+
+# returns (lower, upper) exclusive bounds for the integer values
+# that can be stored into a float
+def getFtoIBounds(floatBits, intBits, signed):
+  sigBits = floating_point_bits_to_type()[floatBits].explicit_significand_bits
+  if not signed:
+    return (-1, 1 << intBits)
+  upper = 1 << intBits - 1
```

Comment:
We might add parentheses here `1 << (intBits - 1)` for clarity, since Python and Swift have different precedence for `<<` and `-`.